[Bug] tools/ptoas/TilelangDaemon.cpp 缺少 getpid / environ 的显式声明，导致在部分 Linux 环境编译失败

### Component

CLI / Tooling (tools/ptoas)

### Description

在部分 Linux 服务器上编译 PTOAS 时，`tools/ptoas/TilelangDaemon.cpp` 会报错：

  ```text
  /xxx/projects/PTOAS/tools/ptoas/TilelangDaemon.cpp:24:53: error: ‘::getpid’ has not been declared; did you mean ‘getpt’?
     24 |   return "/tmp/tilelang_daemon_" + std::to_string(::getpid()) + ".sock";
        |                                                     ^~~~~~
        |                                                     getpt

  /xxx/projects/PTOAS/tools/ptoas/TilelangDaemon.cpp: In static member function ‘static bool ptoas::DaemonManager::start(const string&, const string&, const string&)’:
  /xxx/projects/PTOAS/tools/ptoas/TilelangDaemon.cpp:53:21: error: ‘environ’ was not declared in this scope; did you mean ‘union’?
     53 |     for (char **e = environ; *e; ++e) {
        |                     ^~~~~~~
        |                     union
```

### Reproduction (minimal)

```markdown
TilelangDaemon.cpp 使用了 POSIX 符号：

  - getpid()
  - environ

  但该文件没有显式包含/声明它们：

  - getpid() 一般需要 #include <unistd.h>
  - environ 一般需要显式声明：

    extern char **environ;

  当前代码依赖的是“某些平台头文件的间接声明”。
  因此在一部分机器上可以编过，但在另一部分更严格或头文件实现不同的 Linux 环境上会失败。
```

### Expected behavior

PTOAS 应该能够在不同 Linux 服务器、不同编译器 / libc / 系统头环境下稳定编译，不依赖隐式或间接声明。

### Actual behavior / error logs

```text
构建PTOAS报错
```

### Git commit

dac22d98b50ede7d7b66f6f83b2644a8ac7fec0b

### Host platform

None

### Target Ascend arch (if relevant)

None

### PTOAS build level (if relevant)

None

--- a/tools/ptoas/TilelangDaemon.cpp
+++ b/tools/ptoas/TilelangDaemon.cpp
@@ -12,9 +12,12 @@
 #include "llvm/ADT/SmallVector.h"
 #include "llvm/ADT/StringRef.h"
 #include <chrono>
-#include <thread>
-#include <signal.h>
 #include <cstdlib>
+#include <signal.h>
+#include <thread>
+#include <unistd.h>
+
+extern char **environ;
 
 namespace ptoas {
 

